.github/workflows/publish.yml

```diff
@@ -20,13 +24,41 @@ jobs:
           python-version: "3.x"
 
       - name: "Install dependencies"
-        run: python -m pip install build
+        run: python -m pip install build==0.8.0 sigstore==0.6.2
```

Comment:
Since this is about supply chain security, given PyPI lackluster guarantees, it makes sense to use `pip install --require-hashes` for these? (Using `build-requirements.txt` or such)